src/utils/SparseGrid.h

```diff
@@ -68,7 +69,7 @@ class SparseGridInvasive
      * \return True if and only if an object has been found within the radius.
      */
     bool getNearest(const Point &query_pt, coord_t radius, Elem &elem_nearest,
-                    std::function<bool(const Elem& elem)> precondition = no_precondition) const;
+                    const std::function<bool(const Elem& elem)> precondition = no_precondition) const;
```

Comment:
I don't see a reason this is not passed by const reference, but that was a problem of my original BucketGrid2D...